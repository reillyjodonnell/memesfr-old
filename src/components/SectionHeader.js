import React from 'react';

export default function SectionHeader(props) {
  console.log('Hi');
  return (
    <div data-testid="section-1" className="section-header-area">
      <div data-testid="section-2" className="section-header-title">
        {props.nav === 0 ? (
          <span>Home</span>
        ) : props.nav === 1 ? (
          <span>Trending</span>
        ) : props.nav === 2 ? (
          <span>Popular</span>
        ) : props.nav === 3 ? (
          <span>Recent</span>
        ) : props.nav === 4 ? (
          <span>Random</span>
        ) : null}
      </div>
    </div>
  );
}
