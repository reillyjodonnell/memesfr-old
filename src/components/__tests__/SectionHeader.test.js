import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SectionHeader from '../SectionHeader';

afterEach(() => {
  cleanup();
});

test('should render the SectionHeader', () => {
  render(<SectionHeader />);
  const sectionHeaderElement = screen.getByTestId('section-1');
  expect(sectionHeaderElement).toBeInTheDocument();
});
test('should render Home based on the props passed', () => {
  const nav = 0;
  render(<SectionHeader nav={nav} />);
  const sectionHeaderTitle = screen.getByTestId('section-2');
  expect(sectionHeaderTitle).toBeInTheDocument();
  expect(sectionHeaderTitle).toHaveTextContent('Home');
});
test('should render Trending based on the props passed', () => {
  const nav = 1;
  render(<SectionHeader nav={nav} />);
  const sectionHeaderTitle = screen.getByTestId('section-2');
  expect(sectionHeaderTitle).toBeInTheDocument();
  expect(sectionHeaderTitle).toHaveTextContent('Trending');
});
test('should render Popular based on the props passed', () => {
  const nav = 2;
  render(<SectionHeader nav={nav} />);
  const sectionHeaderTitle = screen.getByTestId('section-2');
  expect(sectionHeaderTitle).toBeInTheDocument();
  expect(sectionHeaderTitle).toHaveTextContent('Popular');
});
test('should render Recent based on the props passed', () => {
  const nav = 3;
  render(<SectionHeader nav={nav} />);
  const sectionHeaderTitle = screen.getByTestId('section-2');
  expect(sectionHeaderTitle).toBeInTheDocument();
  expect(sectionHeaderTitle).toHaveTextContent('Recent');
});
test('should render Random based on the props passed', () => {
  const nav = 4;
  render(<SectionHeader nav={nav} />);
  const sectionHeaderTitle = screen.getByTestId('section-2');
  expect(sectionHeaderTitle).toBeInTheDocument();
  expect(sectionHeaderTitle).toHaveTextContent('Random');
});
test('should match snapshot', () => {
  const nav = 0;
  const tree = renderer.create(<SectionHeader nav={nav} />).toJSON();
  expect(tree).toMatchSnapshot();
});
