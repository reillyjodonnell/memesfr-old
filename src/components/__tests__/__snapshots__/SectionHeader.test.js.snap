// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="section-header-area"
  data-testid="section-1"
>
  <div
    className="section-header-title"
    data-testid="section-2"
  >
    <span>
      Home
    </span>
  </div>
</div>
`;
